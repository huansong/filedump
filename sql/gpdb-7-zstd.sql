-- 1. Tests for pre-generated relation files

-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, compresstype=zstd)
\! pg_filedump -z -M -k -T zstd test/ao.zstd.gp7

-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, orientation=column, compresstype=zstd)
\! pg_filedump -z -M -k -O column -T zstd test/co.zstd.gp7

-- 2. Tests for new relations files generated by the running GPDB server

create table ao_zstd(c1 int, c2 text) with (appendonly=true, compresstype=zstd) distributed replicated;
create table co_zstd(c1 int, c2 text) with (appendonly=true, orientation=column, compresstype=zstd) distributed replicated;
insert into ao_zstd select i,'abc' from generate_series(1,10)i;
insert into co_zstd select i,'abc' from generate_series(1,10)i;
\set relname ao_zstd
\set ext '.1'
\setenv opt '-z -M -k -T zstd'
\ir run_test.sql
\set relname co_zstd
\set ext '.1'
\setenv opt '-z -M -k -O column -T zstd'
\ir run_test.sql

