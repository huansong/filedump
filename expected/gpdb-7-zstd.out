-- 1. Tests for pre-generated relation files
-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, compresstype=zstd)
\! pg_filedump -z -M -k -T zstd test/ao.zstd.gp7

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: test/ao.zstd.gp7
* Options used: -z -M -k -T zstd
*******************************************************************

 Initial Bytes Read = 512

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xff68d373)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 692 compressedLen: 226
 firstRowNumber: 1
Content checksum valid (checksum: 0x0012004e)

***** Block    0 End (Total Block length:  256) ******************
 Initial Bytes Read = 256

***** Block    1 Start ***( Position:  256) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x249741ce)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 692 compressedLen: 226
 firstRowNumber: 101
Content checksum valid (checksum: 0x41d7e725)

***** Block    1 End (Total Block length:  256) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
-- create table <name> (c1 int, c2 text)
--   with (appendonly=true, orientation=column, compresstype=zstd)
\! pg_filedump -z -M -k -O column -T zstd test/co.zstd.gp7

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: test/co.zstd.gp7
* Options used: -z -M -k -O column -T zstd
*******************************************************************

 Initial Bytes Read = 240

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xb2a7c6ee)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 168 compressedLen: 92
 firstRowNumber: 1
Content checksum valid (checksum: 0x6adaf236)

	<AO Content Info> -----
	 Decompressing using "ZSTD" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 38 nullSize: 0 Size: 152
***** Block    0 End (Total Block length:  120) ******************
 Initial Bytes Read = 120

***** Block    1 Start ***( Position:  120) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x531e173f)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 38
 headerLen: 24 uncompressedLen: 168 compressedLen: 92
 firstRowNumber: 101
Content checksum valid (checksum: 0x098081d2)

	<AO Content Info> -----
	 Decompressing using "ZSTD" with compressionLevel: 0
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 38 nullSize: 0 Size: 152
***** Block    1 End (Total Block length:  120) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
-- 2. Tests for new relations files generated by the running GPDB server
create table ao_zstd(c1 int, c2 text) with (appendonly=true, compresstype=zstd) distributed replicated;
create table co_zstd(c1 int, c2 text) with (appendonly=true, orientation=column, compresstype=zstd) distributed replicated;
insert into ao_zstd select i,'abc' from generate_series(1,10)i;
insert into co_zstd select i,'abc' from generate_series(1,10)i;
\set relname ao_zstd
\set ext '.1'
\setenv opt '-z -M -k -T zstd'
\ir run_test.sql
\echo Testing :relname
Testing ao_zstd
checkpoint;
select setting as datadir from gp_settings where gp_segment_id = 0 and name = 'data_directory' \gset
select oid as db_oid from pg_database where datname = current_database() \gset
select relfilenode from gp_dist_random('pg_class') where gp_segment_id = 0 and relname = :'relname' \gset
\set filepath :datadir '/base/' :db_oid '/' :relfilenode :ext
\setenv filepath :filepath
\! pg_filedump $opt $filepath | sed -e "s/logid      ./logid      ./" -e "s/recoff 0x......../recoff 0x......../" -e "s/File: .*/File: /" -e "s/HeapId \(.*\)/HeapId \(\)/" -e "s/IndexId \(.*\)/IndexId \(\)/" -e "s/Checksum: ....../Checksum: /"

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: 
* Options used: -z -M -k -T zstd
*******************************************************************

 Initial Bytes Read = 112

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0xf03ece11)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 10
 headerLen: 24 uncompressedLen: 188 compressedLen: 84
 firstRowNumber: 1
Content checksum valid (checksum: 0x8fb8c7e3)

***** Block    0 End (Total Block length:  112) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
--
----------------------------------------------------------------------------------------------
--
\set relname co_zstd
\set ext '.1'
\setenv opt '-z -M -k -O column -T zstd'
\ir run_test.sql
\echo Testing :relname
Testing co_zstd
checkpoint;
select setting as datadir from gp_settings where gp_segment_id = 0 and name = 'data_directory' \gset
select oid as db_oid from pg_database where datname = current_database() \gset
select relfilenode from gp_dist_random('pg_class') where gp_segment_id = 0 and relname = :'relname' \gset
\set filepath :datadir '/base/' :db_oid '/' :relfilenode :ext
\setenv filepath :filepath
\! pg_filedump $opt $filepath | sed -e "s/logid      ./logid      ./" -e "s/recoff 0x......../recoff 0x......../" -e "s/File: .*/File: /" -e "s/HeapId \(.*\)/HeapId \(\)/" -e "s/IndexId \(.*\)/IndexId \(\)/" -e "s/Checksum: ....../Checksum: /"

*******************************************************************
* PostgreSQL File/Block Formatted Dump Utility
*
* File: 
* Options used: -z -M -k -O column -T zstd
*******************************************************************

 Initial Bytes Read = 80

***** Block    0 Start ***( Position:    0) *************************
<AO Header> -----
 HeaderKind: 1 
 Header Checksum Valid (checksum 0x7318e6e2)
(AoHeaderKind_SmallContent)
 executorBlockKind: 1 rowCount: 10
 headerLen: 24 uncompressedLen: 56 (Block Not Compressed)
 firstRowNumber: 1
Content checksum valid (checksum: 0x47cfd908)

	<AO Content Info> -----
	 DatumStreamVersion_Original  flags: 0
	 ndatum: 10 nullSize: 0 Size: 40
***** Block    0 End (Total Block length:   80) ******************
 Initial Bytes Read = 0

 Done DumpAppendOnlyFileContents.....
--
----------------------------------------------------------------------------------------------
--
